src/main/java/edu/harvard/iq/dataverse/FileDownloadHelper.java

```diff
@@ -149,22 +152,18 @@ public boolean canDownloadFile(FileMetadata fileMetadata){
         // --------------------------------------------------------------------
         
         // --------------------------------------------------------------------
-        // (2) In Dataverse 4.3 and earlier we required that users be authenticated
-        // to download files, but in developing the Private URL feature, we have
-        // added a new subclass of "User" called "PrivateUrlUser" that returns false
-        // for isAuthenticated but that should be able to download restricted files
-        // when given the Member role (which includes the DownloadFile permission).
-        // This is consistent with how Builtin and Shib users (both are
-        // AuthenticatedUsers) can download restricted files when they are granted
-        // the Member role. For this reason condition 2 has been changed. Previously,
-        // we required isSessionUserAuthenticated to return true. Now we require
-        // that the User is not an instance of GuestUser, which is similar in
-        // spirit to the previous check.
+        // (2) A Guest user can only download a restricted file through  membership of an IP Group.
         // --------------------------------------------------------------------
 
         if (session.getUser() instanceof GuestUser){
```

Comment:
Is there a need to treat guest user any differently? Just below, there is a call to check permissions (a call to a method, doesSessionUserHavePermission. I believe we should be able to change that method to use:
this.permissionService.On(objectToCheck).has(permissionToCheck);

instead of: 

this.permissionService.userOn(this.session.getUser(), objectToCheck).has(permissionToCheck);

This would be a nice code cleanup.

Reply:
The discussion continues about this permission check at at https://github.com/IQSS/dataverse/issues/4268#issuecomment-351256372